cvmfs/server/cvmfs_server_transaction.sh

```diff
@@ -78,7 +95,34 @@ cvmfs_server_transaction() {
     # the cvmfs_swissknife lease command needs to be used to acquire a new lease
     if [ x"$upstream_type" = xgw ]; then
         local repo_services_url=$(echo $upstream_storage | cut -d',' -f3)
-        __swissknife lease -a acquire -u $repo_services_url -k $gw_key_file -p $name"/"$subpath || { echo "Could not acquire a new lease for repository $name"; retcode=1; continue; }
+
+        set +e
+
+        local res=1
+        local r=1
+        local delay=$init_retry_delay
+        while true ; do
+          __swissknife lease -a acquire -u $repo_services_url -k $gw_key_file -p $name"/"$subpath
+          res=$?
+          if [ $res -eq 0 ] || [ $retry -eq 0 ] || [ $r -gt $num_retries ]; then
+            break
+          fi
+          echo "Retry $r/$num_retries, waiting: $delay"
+          r=$((r + 1))
+          sleep $delay
+          delay=$((delay * 2))
```

Comment:
I think that needs to be `delay=$(($delay * 2))`

Reply:
I think "$" isn't needed inside double parantheses. I mean, I also checked that it works.

Reply:
Cool, I didn't know this is valid shell.

Reply:
I always have to Google parantheses, brackets, operators etc. in shell!